ui/handler.go

```diff
@@ -121,6 +121,94 @@ func (h *internalHandler) handleNewRulesetRequest(w http.ResponseWriter, r *http
 
 }
 
+func (h *internalHandler) handleEditRulesetRequest(w http.ResponseWriter, r *http.Request) {
+	path := strings.TrimPrefix(r.URL.Path, "/rulesets/")
+	if path == "" {
+		writeError(w, r, nil, http.StatusNotFound)
+		return
+	}
+
+	nrr := &newRulesetRequest{}
+
+	err := json.NewDecoder(r.Body).Decode(nrr)
+	if err != nil {
+		writeError(w, r, err, http.StatusInternalServerError)
+		return
+	}
+
+	// Get the existing entry
+	entry, err := h.service.Get(r.Context(), path, "")
+	if err != nil {
+		if err == store.ErrNotFound {
+			writeError(w, r, err, http.StatusNotFound)
+			return
+		}
+
+		writeError(w, r, err, http.StatusInternalServerError)
+		return
+	}
+
+	// Update the entry with the new rules
+	err, status := updateEntry(entry, nrr)
+	if err != nil {
+		writeError(w, r, err, status)
+		return
+	}
+
+	// Write the new entry back to the DB
+	result, err := h.service.Put(r.Context(), path, entry.Ruleset)
+	if err != nil {
+		if err == store.ErrNotFound {
```

Comment:
The `Put` method will never return the `store.ErrNotFound` error. 
Only `store.ErrNotModified` can be returned.

Reply:
When saying "Only", I mean only ErrNotModified can be returned from our internal types error.

Reply:
You are correct.  Thanks, good catch.